> False positive?
> 
> https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/48081e473e296688e8ba784b355109695448ae73_2026172707/reports/diff/checkstyle/index.html#A16
> 
> ```
> 50    // violation 2 lines below 'Javadoc comment is placed in the wrong location.'
> 51    @Deprecated
> 52    /** Odd javadoc */
> 53    int field22;
> ```

@mahfouz72 @romani No, it is correct. It is not a valid javadoc position.

```
gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % cat Test.java
import java.util.List;

public class Test {

    // violation 2 lines below 'Javadoc comment is placed in the wrong location.'
    @Deprecated
    /** Odd javadoc */
    int field22;

}

gianmarco@gm4 ~/Documents/Projects/testing-checkstyle % javadoc -d docs -private Test.java
Loading source file Test.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 21.0.8+9-LTS
Building tree for all the packages and classes...
Generating docs/Test.html...
Test.java:3: warning: no comment
public class Test {
       ^
Test.java:8: warning: no comment
    int field22;
        ^
Test.java:3: warning: use of default constructor, which does not provide a comment
public class Test {
       ^
Generating docs/package-summary.html...
Copying file ./doc-files/image.png to directory docs/doc-files...
Generating docs/package-tree.html...
Generating docs/overview-tree.html...
Generating docs/deprecated-list.html...
Building index for all classes...
Generating docs/allclasses-index.html...
Generating docs/allpackages-index.html...
Generating docs/index-all.html...
Generating docs/search.html...
Generating docs/index.html...
Generating docs/help-doc.html...
3 warnings
```

<img width="1071" height="245" alt="image" src="https://github.com/user-attachments/assets/e67e0621-fc41-4789-9eec-c9a568150649" />

--- a/config/suppressions.xml
+++ b/config/suppressions.xml
@@ -138,8 +138,6 @@
 
   <!-- until #19146 -->
   <suppress id="noUsageOfGetFileContentsMethod" files="JavadocTypeCheck.java"/>
-  <!-- until #19147 -->
-  <suppress id="noUsageOfGetFileContentsMethod" files="JavadocVariableCheck.java"/>
   <!-- permanent suppression to allow regexp on content of the line -->
   <suppress id="noUsageOfGetFileContentsMethod" files="RegexpCheck.java"/>
   <suppress id="noUsageOfGetFileContentsMethod" files="RegexpSinglelineJavaCheck.java"/>

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocVariableCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocVariableCheck.java
@@ -25,11 +25,10 @@ import java.util.regex.Pattern;
 import com.puppycrawl.tools.checkstyle.StatelessCheck;
 import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
 import com.puppycrawl.tools.checkstyle.api.DetailAST;
-import com.puppycrawl.tools.checkstyle.api.FileContents;
-import com.puppycrawl.tools.checkstyle.api.TextBlock;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 import com.puppycrawl.tools.checkstyle.checks.naming.AccessModifierOption;
 import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
 import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
 import com.puppycrawl.tools.checkstyle.utils.UnmodifiableCollectionUtil;
 
@@ -97,6 +96,11 @@ public class JavadocVariableCheck
     }
 
     @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
     public int[] getDefaultTokens() {
         return getAcceptableTokens();
     }
@@ -120,16 +124,11 @@ public class JavadocVariableCheck
         };
     }
 
-    // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/19147
     @Override
-    @SuppressWarnings("deprecation")
     public void visitToken(DetailAST ast) {
         if (shouldCheck(ast)) {
-            final FileContents contents = getFileContents();
-            final TextBlock textBlock =
-                contents.getJavadocBefore(ast.getLineNo());
-
-            if (textBlock == null) {
+            final DetailAST blockCommentNode = JavadocUtil.getAttachedJavadocComment(ast);
+            if (blockCommentNode == null) {
                 log(ast, MSG_JAVADOC_MISSING);
             }
         }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java
@@ -202,7 +202,7 @@ public final class JavadocUtil {
     public static DetailAST getAttachedJavadocComment(final DetailAST ast) {
         DetailAST result = null;
         DetailAST child = ast.getFirstChild();
-        while (result == null && child != null && !isDeclarationBody(child)) {
+        while (result == null && child.getType() != TokenTypes.IDENT) {
             result = findJavadocComment(child);
             child = child.getNextSibling();
         }
@@ -229,17 +229,6 @@ public final class JavadocUtil {
             }
         }
         return result;
-    }
-
-    /**
-     * Checks whether the node starts a declaration body.
-     *
-     * @param ast the AST node to check
-     * @return {@code true} when the node starts a declaration body
-     */
-    private static boolean isDeclarationBody(DetailAST ast) {
-        final int tokenType = ast.getType();
-        return tokenType == TokenTypes.SLIST;
     }
 
     /**

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocVariableCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocVariableCheckTest.java
@@ -96,6 +96,8 @@ public class JavadocVariableCheckTest
             "15:5: " + getCheckMessage(MSG_JAVADOC_MISSING),
             "22:5: " + getCheckMessage(MSG_JAVADOC_MISSING),
             "41:5: " + getCheckMessage(MSG_JAVADOC_MISSING),
+            "77:38: " + getCheckMessage(MSG_JAVADOC_MISSING),
+            "77:47: " + getCheckMessage(MSG_JAVADOC_MISSING),
         };
         verifyWithInlineConfigParser(
                 getPath("InputJavadocVariableTagsEnums.java"), expected);
@@ -426,6 +428,19 @@ public class JavadocVariableCheckTest
         };
         verifyWithInlineConfigParser(
             getPath("InputJavadocVariableAboveComment.java"),
+            expected);
+    }
+
+    @Test
+    public void testJavadocVariableOddCases() throws Exception {
+        final String[] expected = {
+            "15:5: " + getCheckMessage(MSG_JAVADOC_MISSING),
+            "21:9: " + getCheckMessage(MSG_JAVADOC_MISSING),
+            "21:18: " + getCheckMessage(MSG_JAVADOC_MISSING),
+            "43:9: " + getCheckMessage(MSG_JAVADOC_MISSING),
+        };
+        verifyWithInlineConfigParser(
+            getPath("InputJavadocVariableOddCases.java"),
             expected);
     }
 

--- a/src/test/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtilTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtilTest.java
@@ -22,6 +22,7 @@ package com.puppycrawl.tools.checkstyle.utils;
 import static com.google.common.truth.Truth.assertWithMessage;
 import static com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheck.MSG_EXPECTED_TAG;
 import static com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheck.MSG_RETURN_EXPECTED;
+import static com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocVariableCheck.MSG_JAVADOC_MISSING;
 import static com.puppycrawl.tools.checkstyle.internal.utils.TestUtil.getExpectedThrowable;
 import static com.puppycrawl.tools.checkstyle.internal.utils.TestUtil.isUtilsClassHasPrivateConstructor;
 
@@ -39,6 +40,7 @@ import com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMethodCheck;
 import com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTag;
 import com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTagInfo;
 import com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTags;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocVariableCheck;
 
 public class JavadocUtilTest extends AbstractModuleTestSupport {
 
@@ -459,6 +461,30 @@ public class JavadocUtilTest extends AbstractModuleTestSupport {
         };
         verifyWithInlineConfigParser(
                 getPath("InputJavadocUtilCompactCtorDefComments.java"), expected);
+    }
+
+    @Test
+    public void testGetAttachedJavadocCommentForVariableDefinitions() throws Exception {
+        final String[] expected = {
+            "17:5: " + getCheckMessage(JavadocVariableCheck.class, MSG_JAVADOC_MISSING),
+            "20:5: " + getCheckMessage(JavadocVariableCheck.class, MSG_JAVADOC_MISSING),
+            "24:5: " + getCheckMessage(JavadocVariableCheck.class, MSG_JAVADOC_MISSING),
+            "32:5: " + getCheckMessage(JavadocVariableCheck.class, MSG_JAVADOC_MISSING),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputJavadocUtilVariableDefComments.java"), expected);
+    }
+
+    @Test
+    public void testGetAttachedJavadocCommentForEnumConstantDefinitions()
+            throws Exception {
+        final String[] expected = {
+            "21:5: " + getCheckMessage(JavadocVariableCheck.class, MSG_JAVADOC_MISSING),
+            "28:5: " + getCheckMessage(JavadocVariableCheck.class, MSG_JAVADOC_MISSING),
+            "31:5: " + getCheckMessage(JavadocVariableCheck.class, MSG_JAVADOC_MISSING),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputJavadocUtilEnumConstantDefComments.java"), expected);
     }
 
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableOddCases.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableOddCases.java
@@ -1,0 +1,52 @@
+/*
+JavadocVariable
+accessModifiers = (default)public,protected,package,private
+ignoreNamePattern = (default)null
+tokens = VARIABLE_DEF, ENUM_CONSTANT_DEF
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;
+
+
+class InputJavadocVariableOddCases {
+
+    // violation below 'Missing a Javadoc comment'
+    @Deprecated
+    /** javadoc tool does not consider this as javadoc comment, so checkstyle does the same */
+    int field22;
+
+    /** Javadoc */
+    private static enum PathStatus {
+        INVALID, CHECKED }; // 2 violations
+
+    // the following is taken
+    // from openjdk25/src/java.base/share/classes/jdk/internal/vm/Continuation.java
+
+    public enum Pinned {
+        /** Native frame on stack */ NATIVE,
+        /** Monitor held */ MONITOR,
+        /** In critical section */ CRITICAL_SECTION,
+        /** Exception (OOME/SOE) */ EXCEPTION
+    }
+
+    /** Preemption attempt result */
+    public enum PreemptStatus {
+        /** Success */                                                      SUCCESS(null),
+        /** Permanent failure */                                            PERM_FAIL_U(null),
+        /** Permanent failure: continuation already yielding */             PERM_FAIL_Y(null),
+        /** Permanent failure: continuation not mounted on the thread */    PERM_FAIL_N(null),
+        /** Transient failure: due to a held CS */      CRITICAL_SECTION(Pinned.CRITICAL_SECTION),
+        /** Transient failure: due to native frame */   NATIVE(Pinned.NATIVE),
+        /** Transient failure: due to a held monitor */ MONITOR(Pinned.MONITOR);
+
+        final Pinned pinned; // violation 'Missing a Javadoc comment'
+        private PreemptStatus(Pinned reason) { this.pinned = reason; }
+        /**
+         * Whether or not the continuation is pinned.
+         * @return whether or not the continuation is pinned
+         **/
+         public Pinned pinned() { return pinned; }
+    }
+
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableTagsEnums.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableTagsEnums.java
@@ -69,6 +69,12 @@ public class InputJavadocVariableTagsEnums {
     {
         return null;
     }
+
+    // 2 violations 4 lines below:
+    //      'Missing a Javadoc comment'
+    //      'Missing a Javadoc comment'
+    /** Javadoc */
+    private static enum PathStatus { INVALID, CHECKED };
 }
 
 /**
@@ -77,4 +83,3 @@ public class InputJavadocVariableTagsEnums {
 class WrongException extends RuntimeException
 {
 }
-

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/utils/javadocutil/InputJavadocUtilEnumConstantDefComments.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/utils/javadocutil/InputJavadocUtilEnumConstantDefComments.java
@@ -1,0 +1,32 @@
+/*
+com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocVariableCheck
+accessModifiers = (default)public,protected,package,private
+ignoreNamePattern = (default)null
+tokens = (default)VARIABLE_DEF, ENUM_CONSTANT_DEF
+
+*/
+
+package com.puppycrawl.tools.checkstyle.utils.javadocutil;
+
+enum InputJavadocUtilEnumConstantDefComments {
+
+    /**constant*/
+    FIRST,
+
+    /**annotation*/
+    @Deprecated
+    ANNOTATED,
+
+    // violation below 'Missing a Javadoc comment'
+    BODY {
+        /**nope*/
+        void method() {
+        }
+    },
+
+    // violation below 'Missing a Javadoc comment'
+    NO_JAVADOC,
+
+    // violation below 'Missing a Javadoc comment'
+    REAL;
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/utils/javadocutil/InputJavadocUtilVariableDefComments.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/utils/javadocutil/InputJavadocUtilVariableDefComments.java
@@ -1,0 +1,37 @@
+/*
+com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocVariableCheck
+accessModifiers = (default)public,protected,package,private
+ignoreNamePattern = (default)null
+tokens = VARIABLE_DEF
+
+*/
+
+package com.puppycrawl.tools.checkstyle.utils.javadocutil;
+
+class InputJavadocUtilVariableDefComments {
+
+    /**field*/
+    int field;
+
+    // violation below 'Missing a Javadoc comment'
+    private int modifierPath;
+
+    // violation below 'Missing a Javadoc comment'
+    @Deprecated
+    String annotationPath;
+
+    // violation below 'Missing a Javadoc comment'
+    int noJavadoc;
+
+    /**dangling*/
+
+    /**real*/
+    String danglingReal;
+
+    // violation below 'Missing a Javadoc comment'
+    Object initializerCommentOnly = new Object() {
+        /**nope*/
+        class Nested {
+        }
+    };
+}

